src/vm/eventpipebuffermanager.cpp

```diff
@@ -1050,19 +1050,30 @@ void EventPipeBufferManager::SuspendWriteEvent(uint32_t sessionIndex)
 
     // Wait for any straggler WriteEvent threads that may have already allocated a buffer but
     // hadn't yet relinquished it.
-    {
+    bool hasWriteInProgress = false;
+    do {
+        // check whether we are breaking out of the loop due a blocked write. If we are, then sleep arbitrary length before taking m_lock
+        if (hasWriteInProgress)
+            ClrSleepEx(10 /* ms */, FALSE);
+
         SpinLockHolder _slh(&m_lock);
         SListElem<EventPipeThreadSessionState *> *pElem = m_pThreadSessionStateList->GetHead();
         while (pElem != NULL)
         {
+            // initial assumption is that there is a write in progress
+            hasWriteInProgress = true;
```

Comment:
I would set this false before the while loop and avoid setting it to true until immediately prior to 1075 out of caution. I don't think it is possible for the if statements at 1067 or 1070 to evaluate as false, but if I missed something or it ever became possible in the future then this do {} while loop could become an infinite loop. Similarly I don't think it is possible for the m_pThreadSessionStateList to have > 0 elements on the first pass and 0 elements on a later pass, but if that ever became possible this would also degenerate into an infinite loop.